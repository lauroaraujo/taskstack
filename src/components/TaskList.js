import React from 'react';
import { connect } from 'react-redux';
import Task from './Task';

export const TaskList = ({ tasks }) => (
  <ul>
    { tasks.map((t) => renderTask(t)) }
  </ul>
);

function renderTask(task) {
  const {id, status, content} = task;
  return (
    <li key={`t_${id}`}>
      <Task id={id} status={status} content={content} />
    </li>);
}

function mapStateToProps({ taskList }) {
  return {
    tasks: taskList.tasks,
  }
}

export default connect(mapStateToProps)(TaskList);
