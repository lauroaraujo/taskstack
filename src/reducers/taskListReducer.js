import { TASK_ACTIONS } from '../actions/taskActions';

export const DEFAULT_STATE = {
  tasks: [
    { id: 1, status: 'PENDING', content: 'Do the dishes' },
    { id: 2, status: 'BLOCKED', content: 'Use task app' },
    { id: 3, status: 'DONE', content: 'Start task app' },
    { id: 4, status: 'DROPPED', content: 'Games Backlog App' },
  ],
}

const { SET_TASK_STATUS } = TASK_ACTIONS;

export default function tasksReducer(state = DEFAULT_STATE, action) {
  switch(action.type) {
      case SET_TASK_STATUS:
        const tasks = state.tasks.map((t) => {
          const { id, status } = action.payload;
          return (t.id === id) ? {...t, status} : t;
        });
        return { ...state, tasks }
      default:
        return state;
    }
}