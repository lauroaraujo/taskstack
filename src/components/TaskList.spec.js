import React from 'react';
import { shallow } from 'enzyme';
import { DEFAULT_STATE } from '../reducers/taskListReducer';
import { TaskList } from './TaskList';

describe('TaskList component', () => {
  it('renders correctly', () => {
    const list = shallow(<TaskList tasks={[...DEFAULT_STATE.tasks]}/>);
    expect(list).toMatchSnapshot();
  });
});
