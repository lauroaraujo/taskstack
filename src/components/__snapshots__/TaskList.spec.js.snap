// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component renders correctly 1`] = `
<ul>
  <li
    key="t_1"
  >
    <Connect(Task)
      content="Do the dishes"
      id={1}
      status="PENDING"
    />
  </li>
  <li
    key="t_2"
  >
    <Connect(Task)
      content="Use task app"
      id={2}
      status="BLOCKED"
    />
  </li>
  <li
    key="t_3"
  >
    <Connect(Task)
      content="Start task app"
      id={3}
      status="DONE"
    />
  </li>
  <li
    key="t_4"
  >
    <Connect(Task)
      content="Games Backlog App"
      id={4}
      status="DROPPED"
    />
  </li>
</ul>
`;
